src/devtools/pachdev/actions.go

```diff
@@ -0,0 +1,571 @@
+func (op *deployOp) helmChartSource() string {
+	if op.pachVersion == "latest" {
+		// // TODO(msteffen): this codepath has no automated tests. Maybe set up
+		// // mock GitHub? (has worked in the past, and I bet the releases endpoint
+		// // is simple)
+		// client := github.NewClient(nil)
+		// // Not sure if this is the correct call -- want to filter out alphas,
+		// // nightlies, etc.
+		// release, _, err := client.Repositories.GetLatestRelease(context.Background(), "pachyderm", "pachyderm")
+		// if err != nil {
+		// 	panic(err)
+		// }
+		// return fmt.Sprintf("https://github.com/pachyderm/helmchart/releases/download/pachyderm-%s/pachyderm-%s.tgz", *release.TagName, *release.TagName)
+		panic("not implemented")
+	}
+	if op.pachVersion == "nightly" {
+		// look up from dockerhub or something, somehow
+		panic("not implemented")
+	}
+	if op.pachVersion == "local" {
+		return "./etc/helm/pachyderm"
+	}
+	panic("do not know how to get helm chart for pachVersion " + op.pachVersion)
+}
+
+// getHelmArgs is a helper for K8sManifest that returns the arguments it should
+// pass to helm
+func (op *deployOp) getHelmArgs() []string {
+	args := []string{
+		"--set", "pachd.image.tag=local",
+		"--set", "pachd.enterpriseLicenseKey=" + os.Getenv("ENT_ACT_CODE"), // used in CI -- weird name, though
+		"--set", "pachd.lokiDeploy=false",
+		"--set", "pachd.lokiLogging=false",
+		"--set", "pachd.clusterDeploymentID=dev",
+		"--set", "proxy.service.type=NodePort",
+		"--set", "pachd.rootToken=iamroot",
+	}
+
+	if op.objectStorage == "minio" {
+		args = append(args,
+			"--set", "deployTarget=custom",
+			"--set", "pachd.storage.backend=MINIO",
+			"--set", "pachd.storage.minio.bucket=pachyderm-test",
+			"--set", "pachd.storage.minio.endpoint=minio.default.svc.cluster.local:9000",
+			"--set", "pachd.storage.minio.id=minioadmin",
+			"--set", "pachd.storage.minio.secret=minioadmin",
+			"--set-string", "pachd.storage.minio.signature=",
+			"--set-string", "pachd.storage.minio.secure=false",
+		)
+	} else if op.objectStorage == "local" {
+		args = append(args,
+			"--set", "deployTarget=LOCAL",
+		)
+	}
+
+	return args
+}
+
+// K8sManifest is an accessor for op.k8sManifest that lazily initializes the
+// field if it's not already set.  All access to op.k8sManifest should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) K8sManifest() string {
+	if op.err != nil {
+		return ""
+	}
+	if op.k8sManifest != "" {
+		return op.k8sManifest
+	}
+
+	helmArgs := []string{"template", "pach", op.helmChartSource()}
+	helmArgs = append(helmArgs, op.getHelmArgs()...)
+	cmd := exec.Command("helm", helmArgs...)
+	stdout, stderr := &bytes.Buffer{}, &bytes.Buffer{}
+	cmd.Stdout, cmd.Stderr = stdout, stderr
+	if err := cmd.Run(); err != nil {
+		op.err = errors.Wrap(err, "failed to generate k8s manifest with helm:\n"+stderr.String())
+		return ""
+	}
+	op.k8sManifest = stdout.String()
+	return op.k8sManifest
+}
```

Comment:
That works!  I am OK with just shelling out to `helm install` or `helm upgrade` and not touching the manifests in code at all.